src/adapter/src/catalog/builtin.rs

```diff
@@ -1389,16 +1389,13 @@ pub static MZ_AUDIT_EVENTS: Lazy<BuiltinTable> = Lazy::new(|| BuiltinTable {
 
 pub static MZ_SOURCE_STATUS_HISTORY: Lazy<BuiltinSource> = Lazy::new(|| BuiltinSource {
     name: "mz_source_status_history",
-    schema: MZ_CATALOG_SCHEMA,
-    data_source: None,
+    schema: MZ_INTERNAL_SCHEMA,
+    data_source: Some(IntrospectionType::SourceStatusHistory),
     desc: RelationDesc::empty()
         .with_column("timestamp", ScalarType::Timestamp.nullable(false))
-        .with_column("source_name", ScalarType::String.nullable(false))
         .with_column("source_id", ScalarType::String.nullable(false))
-        .with_column("source_type", ScalarType::String.nullable(false))
-        .with_column("upstream_name", ScalarType::String.nullable(true))
-        .with_column("worker_id", ScalarType::Int64.nullable(false))
-        .with_column("worker_count", ScalarType::Int64.nullable(false))
+        .with_column("worker_id", ScalarType::Int64.nullable(true))
+        .with_column("worker_count", ScalarType::Int64.nullable(true))
```

Comment:
This seems like the wrong place for the `worker_count` column, no? Should that not be in `mz_sources` directly?

Reply:
So, yes and no. In particular, the worker count for a source can change over its lifetime - and when it does old logs would become uninterpretable. (We could theoretically add a new log-shaped table for those changes too, and serialize timestamps across the two so they could be queried consistently, but IMO that sounds too complex / heavy for the value it adds.)

A third option is to remove the worker columns entirely, and scope the status per-source instead of per-worker. This is my personal preference: makes this table easier to interpret, and we can add worker to the metadata JSON if we think anyone cares. The cost of this is mostly some additional implementation complexity: the existing `Healthchecker` implementation tracks the health of each specific timely worker, and we'd need to replace it with something that tracks the global health of a source. (A bit more plumbing but quite tractable.)

Reply:
We discussed this elsewhere, and have decided to go with the third option: removing the worker id columns. Cliffs notes:
- We're trying to keep the catalog fully normalized, and normalizing the worker count would be a pain. (For the reason summarized above.)
- The worker count is not meaningful for all states, and we think it's unlikely users will want to query it. (If it turns out to be useful, we'll add it to the details JSON.)
- The additional implementation complexity is not too high, and we're willing to eat some cost there for better UX.